server/bleep/src/agent/transcoder.rs

```diff
@@ -494,45 +494,11 @@ fn xml_for_each(article: &str, f: impl Fn(&str) -> Option<String>) -> String {
 }
 
 fn try_trim_code_xml(xml: &str) -> Result<String> {
-    let xml = fixup_xml_code(xml);
-
-    let code_chunk = quick_xml::de::from_str(&xml).context("couldn't parse as XML code block")?;
-
-    Ok(match code_chunk {
-        CodeChunk::QuotedCode {
-            code: _,
-            language,
-            path,
-            start_line,
-            end_line,
-        } => {
-            let start_line = start_line
-                .map(|n| format!("<StartLine>{n}</StartLine>\n"))
-                .unwrap_or_default();
-            let end_line = end_line
-                .map(|n| format!("<EndLine>{n}</EndLine>\n"))
-                .unwrap_or_default();
-
-            format!(
-                "<QuotedCode>\n\
-                <Code>[REDACTED]</Code>\n\
-                <Language>{language}</Language>\n\
-                <Path>{path}</Path>\n\
-                {start_line}\
-                {end_line}\
-                </QuotedCode>"
-            )
-        }
+    // We just remove code chunks completely.
 
-        CodeChunk::GeneratedCode { code: _, language } => {
-            format!(
-                "<GeneratedCode>\n\
-                <Code>[REDACTED]</Code>\n\
-                <Language>{language}</Language>\n\
-                </GeneratedCode>"
-            )
-        }
-    })
+    let xml = fixup_xml_code(xml);
```

Comment:
Do we need to do this work at all if we're returning an empty string?

Reply:
We do need to do this for partially generated messages. E.g. if a message generates with `<GeneratedCode><Code>foo`, we need to call `fixup_xml_code` to complete the block so that it can still be parsed on the line right below. If parsing fails here (e.g. there is a *different* kind of XML block, perhaps HTML in the markdown), we know the block is not a code chunk and should be kept.